contributors/design-proposals/bootstrap-discovery.md

```diff
@@ -173,6 +171,44 @@ In addition, `jws-kubeconfig-<token-id>-hash` will be set to the MD5 hash of the
 
 This controller will also delete tokens that are past their expiration time.
 
+## `kubeadm` UX
+
+We extend kubeadm with a set of flags and helper commands for managing and using these tokens.
+
+### `kubeadm init` flags
+
+* `--token` If set, this injects the bootstrap token to use when initializing the cluster.  If this is unset, then a random token is created and shown to the user.  If set explicitly to the empty string then no token is generated or created.
+* `--token-ttl` If set, this sets the TTL for the lifetime of this token.  Defaults to 0 which means "forever"
+* `--discovery-token` If set, this is a token that is created in the new cluster that is used for signing discovery bootstrap.  If not set, it defaults to the value from `--token`.  If set to the empty string, there is no token created.
+* `--discovery-token-ttl` Similar TTL for token.
+* `--bootstrap-auth-token` (not strictly part of this proposal) If set, this token is set into the cluster to be used for doing authentication for TLS bootstrap.  If set to the empty string there is no token created.
+* `--boostrap-auth-token-ttl`
+
+### `kubeadm join` flags
+
+* `--token` This sets the token for both discovery and bootstrap auth.
+* `--discovery-url` If set this will grab the cluster-info data (a kubeconfig) from a URL.  This will support `http` and `https+insecure` but it we will warn if this is not `https`.  This also supports `username:password@host` syntax for doing HTTP auth.
+* `--discovery-file` If set, this will load the cluster-info from a file.
+* `--discovery-token` If set, (or set via `--token`) then we will be using the token scheme described above.
+* `--discovery-token-url` This is the URL to hit to get the ConfigMap that contains the kubeconfig and signatures.  This defaults to `https+insecure://<api-server>/api/v1/namespaces/kube-public/configmaps/cluster-info`.
```

Comment:
I don't think this is needed. I've implemented this as a `client.CoreV1().ConfigMaps(metav1.NamespacePublic).Get(bootstrapapi.ClusterInfoConfigMap)` and I don't think this path should be configurable.
If the user needs to config this, go with https or file